movement/movement.c

```diff
@@ -627,29 +634,53 @@ static movement_event_type_t _figure_out_button_event(bool pin_level, movement_e
         // now that that's out of the way, handle falling edge
         uint16_t diff = movement_state.fast_ticks - *down_timestamp;
         *down_timestamp = 0;
-        _movement_disable_fast_tick_if_possible();
         // any press over a half second is considered a long press. Fire the long-up event
         if (diff > MOVEMENT_LONG_PRESS_TICKS) return button_down_event_type + 3;
         else return button_down_event_type + 1;
     }
 }
 
-void cb_light_btn_interrupt(void) {
-    bool pin_level = watch_get_pin_level(BTN_LIGHT);
+static void light_btn_action(bool pin_level) {
     _movement_reset_inactivity_countdown();
     event.event_type = _figure_out_button_event(pin_level, EVENT_LIGHT_BUTTON_DOWN, &movement_state.light_down_timestamp);
 }
 
-void cb_mode_btn_interrupt(void) {
-    bool pin_level = watch_get_pin_level(BTN_MODE);
+static void mode_btn_action(bool pin_level) { 
     _movement_reset_inactivity_countdown();
     event.event_type = _figure_out_button_event(pin_level, EVENT_MODE_BUTTON_DOWN, &movement_state.mode_down_timestamp);
 }
 
-void cb_alarm_btn_interrupt(void) {
-    bool pin_level = watch_get_pin_level(BTN_ALARM);
+static void alarm_btn_action(bool pin_level) {
     _movement_reset_inactivity_countdown();
-    event.event_type = _figure_out_button_event(pin_level, EVENT_ALARM_BUTTON_DOWN, &movement_state.alarm_down_timestamp);
+    uint8_t event_type = _figure_out_button_event(pin_level, EVENT_ALARM_BUTTON_DOWN, &movement_state.alarm_down_timestamp);
+    if  (movement_state.ignore_alarm_btn_after_sleep){
+        if (event_type == EVENT_ALARM_BUTTON_UP || event_type == EVENT_ALARM_LONG_UP) movement_state.ignore_alarm_btn_after_sleep = false;
+        return;
+    }
```

Comment:
This branch could be dead code eliminated if the boolean variable is converted to compile time configuration with macros.

```C
    if  (MOVEMENT_IGNORE_ALARM_AFTER_SLEEP) {
        // ...
    }
```

If not configured to do that, it becomes:

```C
    if  (0) {
        // ...
    }
```

And the compiler eliminates the entire branch.

Reply:
I agree, if we want to make the variable a preprocessor step

Reply:
I think it's a good idea to make it a preprocessor macro because I don't expect that users would want to change this setting at runtime. The discussions I see on discord appear to agree with this view point. It makes code more efficient and still allows people to revert to the previous behavior if needed.

Reply:
`ignore_alarm_btn_after_sleep` isn't asetting, it's a flag that goes high when we enter sleep and gets reset here.

Reply:
Oh. My mistake, looks like I misunderstood its purpose. Please ignore all discussion regarding it.